tests/components/samsungtv/test_media_player.py

```diff
@@ -55,247 +87,322 @@ class UnhandledResponse(Exception):
-        self.device.send_key("HELLO")
-        self.assertIsNone(self.device._remote)
-        self.assertEqual(STATE_ON, self.device._state)
-
-    def test_send_key_os_error(self):
-        """Testing broken pipe Exception."""
-        _remote = mock.Mock()
-        _remote.control = mock.Mock(side_effect=OSError("Boom"))
-        self.device.get_remote = mock.Mock(return_value=_remote)
-        self.device.send_key("HELLO")
-        assert self.device._remote is None
-        assert STATE_OFF == self.device._state
-
-    def test_power_off_in_progress(self):
-        """Test for power_off_in_progress."""
-        assert not self.device._power_off_in_progress()
-        self.device._end_of_power_off = dt_util.utcnow() + timedelta(seconds=15)
-        assert self.device._power_off_in_progress()
-
-    def test_name(self):
-        """Test for name property."""
-        assert "fake" == self.device.name
-
-    def test_state(self):
-        """Test for state property."""
-        self.device._state = STATE_ON
-        self.assertEqual(STATE_ON, self.device.state)
-        self.device._state = STATE_OFF
-        assert STATE_OFF == self.device.state
-
-    def test_is_volume_muted(self):
-        """Test for is_volume_muted property."""
-        self.device._muted = False
-        assert not self.device.is_volume_muted
-        self.device._muted = True
-        assert self.device.is_volume_muted
-
-    def test_supported_features(self):
-        """Test for supported_features property."""
-        self.device._mac = None
-        assert SUPPORT_SAMSUNGTV == self.device.supported_features
-        self.device._mac = "fake"
-        assert SUPPORT_SAMSUNGTV | SUPPORT_TURN_ON == self.device.supported_features
-
-    def test_turn_off(self):
-        """Test for turn_off."""
-        self.device.send_key = mock.Mock()
-        _remote = mock.Mock()
-        _remote.close = mock.Mock()
-        self.get_remote = mock.Mock(return_value=_remote)
-        self.device._end_of_power_off = None
-        self.device.turn_off()
-        assert self.device._end_of_power_off is not None
-        self.device.send_key.assert_called_once_with("KEY_POWER")
-        self.device.send_key = mock.Mock()
-        self.device._config["method"] = "legacy"
-        self.device.turn_off()
-        self.device.send_key.assert_called_once_with("KEY_POWEROFF")
-
-    @mock.patch("homeassistant.components.samsungtv.media_player._LOGGER.debug")
-    def test_turn_off_os_error(self, mocked_debug):
-        """Test for turn_off with OSError."""
+@pytest.fixture
+def samsung_mock():
+    """Mock samsungctl."""
+    with MockDependency("samsungctl"):
+        yield
+
+
+@pytest.fixture
+def wakeonlan_mock():
+    """Mock wakeonlan."""
+    with MockDependency("wakeonlan"):
+        yield
+
+
+async def setup_samsungtv(hass, config):
+    """Set up mock Samsung TV."""
+    with mock.patch("homeassistant.components.samsungtv.media_player.socket"):
```

Comment:
Maybe it's safer to move this patch to a pytest fixture so that it's applied during the whole test?